dds/DCPS/Observer.h

```diff
@@ -0,0 +1,122 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_OBSERVER_H
+#define OPENDDS_DCPS_OBSERVER_H
+
+#include "RcObject.h"
+#include "Definitions.h"
+#include "SequenceNumber.h"
+
+#include <dds/DdsDcpsCoreC.h>
+#include <dds/DdsDcpsPublicationC.h>
+#include <dds/DdsDcpsSubscriptionC.h>
+#include <dds/DdsDcpsInfoUtilsC.h>
+
+#if !defined (ACE_LACKS_PRAGMA_ONCE)
+#pragma once
+#endif /* ACE_LACKS_PRAGMA_ONCE */
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+class DataSampleElement;
+class ReceivedDataSample;
+class ReceivedDataElement;
+
+class OpenDDS_Dcps_Export Observer
+  : public virtual RcObject
+{
+public:
+  typedef Observer* Ptr;
+  typedef TAO_Objref_Var_T<Observer> Var;
+  typedef TAO_Objref_Out_T<Observer> Out;
```

Comment:
Are these used?

Reply:
At this high level interface, it's probably a good thing to provide these commonly used definitions. Currently these three of the four definitions are only used for defining their DDS-spec-complied aliases.

Reply:
The `_var`, `_ptr`, and `_out` types are from IDL-to-C++.  There is no IDL here.  Let's remove these.